test/plugin/in_tail/test_io_handler.rb

```diff
@@ -146,5 +146,36 @@ def create_watcher
       assert_equal 5, returned_lines[0].size
       assert_equal 3, returned_lines[1].size
     end
+
+    test 'call receive_lines some times when long line(more than 8192) and max_line_size is 4096' do
```

Comment:
I would like this test to be more clear about what is being tested.

This test does not seem to be intended to test if `receive_lines` is called more than once (some times).
(Actually, it is called only once in this test.)

Perhaps it would be better to make this test independent of the `when limit is 5` sub test.

Reply:
Hi @daipom 

What about the test code below?

```
  test 'does not call receive_lines when line_size exceeds max_line_size' do
    t = 'x' * (8192)
    text = "#{t}\n"

    max_line_size = 8192

    @file.write(text)
    @file.close

    update_pos = 0

    watcher = create_watcher
    stub(watcher).pe do
      pe = 'position_file'
      stub(pe).read_pos {0}
      stub(pe).update_pos { |val| update_pos = val }
      pe
    end

    returned_lines = []
    r = Fluent::Plugin::TailInput::TailWatcher::IOHandler.new(watcher, path: @file.path, read_lines_limit: 1000, read_bytes_limit_per_second: -1, max_line_size: max_line_size, log: $log, open_on_every_update: false, metrics: @metrics) do |lines, _watcher|
      returned_lines << lines.dup
      true
    end

    r.on_notify
    assert_equal text.bytesize, update_pos
    assert_equal 0, returned_lines.size
  end
```

I put this test outside of the 'when limit is 5' subtest and changed its name and behavior.

But I discovered that this test failed because it doesn't not call update_pos.
So, we need to fix the code. I will push one more commit for it.

Reply:
@yugeeklab Thanks for considering the test code!
Basically, it looks good, but we need to be careful about how we handle `pos`.

> But I discovered that this test failed because it doesn't not call update_pos.
> So, we need to fix the code. I will push one more commit for it.

I saw bf73efeea722376f30bccac0f10095fb6601939c and realized that there is a problem that needs to be solved about the management of `pos`
I'll comment on it in the new thread.